hapi-fhir-validation/src/test/java/org/hl7/fhir/common/hapi/validation/support/InMemoryTerminologyServerValidationSupportTest.java

```diff
@@ -423,8 +423,45 @@ public void testExpandValueSet_VsUsesVersionedSystem_CsIsCompleteWithCode() {
 		assertEquals("MODERNA COVID-19 mRNA-1273", valueSet.getExpansion().getContains().get(0).getDisplay());
 	}
 
-
-	private static class PrePopulatedValidationSupportDstu2 extends PrePopulatedValidationSupport {
+    @Test
+    void testValidateCodeWhenCodeSystemUrlIsInCanonicalFormat() {
+		 ConceptValidationOptions options = new ConceptValidationOptions();
+		 ValidationSupportContext valCtx = new ValidationSupportContext(myChain);
+
+		 String pipedCodeSystemUrl = "http://terminology.hl7.org/CodeSystem/v2-0360|2.7";
+		 String codeMD = "MD";
+
+		 CodeSystem cs = new CodeSystem();
+		 cs.setStatus(Enumerations.PublicationStatus.ACTIVE);
+		 cs.setContent(CodeSystem.CodeSystemContentMode.COMPLETE);
+		 cs.setUrl(pipedCodeSystemUrl);
+		 cs.setVersion("2.7");
+		 cs.addConcept()
+			 .setCode(codeMD)
+			 .setDisplay("Doctor of Medicine");
+		 myPrePopulated.addCodeSystem(cs);
+
+		 ValueSet theValueSet = new ValueSet();
+		 theValueSet.setUrl("http://someValueSetURL");
+		 theValueSet.setVersion("0360");
+		 theValueSet.getCompose().addInclude().setSystem(pipedCodeSystemUrl);
+
+		 String theCodeToValidateCodeSystemUrl = pipedCodeSystemUrl;
+		 String theCodeToValidate = codeMD;
+
+		 IValidationSupport.CodeValidationResult codeValidationResult = mySvc.validateCodeInValueSet(
+			 valCtx,
+			 options,
+			 theCodeToValidateCodeSystemUrl,
+			 theCodeToValidate,
+			 "dontCare",
```

Comment:
```suggestion
			 null,
```
i noticed the other tests used `null`, would `null` be better?

Reply:
In this case, it would not matter and i can change it for consistency.  But as a general rule, passing a NULL value implies that the implementer of the functionality assumes that the parameter can be NULL and will code accordingly.  In such case, I would expect an @Nullable annotation on the parameter or have it wrapped in an Optional<T>.  That is not the case here so as a rule of thumb, i always provide a filler value since i assume that the inner working of the method will dereference the point without asserting that it is non-null.